shared/js/background/onboarding.es6.js

```diff
@@ -0,0 +1,115 @@
+/**
+* This is only injected on duckduckgo.com to assist with user onboarding
+*/
+
+function createOnboardingCode ({
+    isAddressBarQuery,
+    showWelcomeBanner,
+    showCounterMessaging,
+    extensionId,
+    browser
+} = {}) {
+    const params = JSON.stringify({
+        isAddressBarQuery,
+        showWelcomeBanner,
+        showCounterMessaging,
+        extensionId,
+        browser
+    })
+
+    // TODO: upgrade to `chrome.scripting.executeScript` when we upgrade to manifest v3
+    // as it allows to inject a function with _arguments_. Here we simulate that in a hacky way
+    return `(${onboarding.toString()})(${params})`
+}
+
+function onboarding ({
+    isAddressBarQuery,
+    showWelcomeBanner,
+    showCounterMessaging,
+    browser,
+    extensionId
+}) {
+    function getDocumentStartData (cb) {
```

Comment:
This function could be made more JavaScript like by using a promise instead:
```
function getDocumentStartData (cb) {
    if (browser !== 'chrome') {
        return null
    }
    return new Promise((resolve) => {
        window.postMessage({ type: 'documentStartDataRequest' }, window.location.origin)
            window.addEventListener('message', function handleMessage (e) {
                if (e.origin === window.location.origin && e.data.type === 'documentStartDataResponse') {
                    window.removeEventListener('message', handleMessage)
                    resolve(e.data.payload)
                }
            })
        }
    })
}
```

Then usage:
```
const documentStartData = await getDocumentStartData();
```

Reply:
I have kept a callback (instead of a Promise + async/await) to avoid needing to import 'regenerator-runtime/runtime' until we configure `@babel/preset-env` more precisely. I think that this is OK for now but I can refactor to Promises + async/await if you feel strongly about it.